cursorless-talon/src/targets/range_target.py

```diff
@@ -0,0 +1,67 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from talon import Module
+
+from .target_types import ImplicitTarget, PrimitiveTarget, RangeTarget
+
+mod = Module()
+
+mod.list(
+    "cursorless_range_connective",
+    desc="A range joiner that indicates whether to include or exclude anchor and active",
+)
+
+
+@dataclass
+class RangeConnective:
+    excludeAnchor: bool
+    excludeActive: bool
+
+
+@dataclass
+class RangeConnectiveWithType:
+    connective: RangeConnective
+    type: Optional[str]
+
+
+@mod.capture(rule="{user.cursorless_range_connective}")
+def cursorless_range_connective(m) -> RangeConnective:
+    return RangeConnective(
+        m.cursorless_range_connective in ["rangeExclusive", "rangeExcludingStart"],
+        m.cursorless_range_connective in ["rangeExclusive", "rangeExcludingEnd"],
+    )
+
+
+@mod.capture(
+    rule="[<user.cursorless_range_type>] <user.cursorless_range_connective> | <user.cursorless_range_type>"
+)
+def cursorless_range_connective_with_type(m) -> RangeConnectiveWithType:
+    return RangeConnectiveWithType(
+        getattr(m, "cursorless_range_connective", RangeConnective(False, False)),
+        getattr(m, "cursorless_range_type", None),
+    )
+
+
+@mod.capture(
+    rule=(
+        "<user.cursorless_range_connective_with_type> <user.cursorless_primitive_target> | "
+        "<user.cursorless_primitive_target> <user.cursorless_range_connective_with_type> <user.cursorless_primitive_target>"
```

Comment:
how come we can't unify these to
```suggestion
        "[<user.cursorless_primitive_target>] <user.cursorless_range_connective_with_type> <user.cursorless_primitive_target>"
```